### ISSUE TITLE
MultiJson + JrJackson + JRuby-Rack: Can't cast RackInput to RubyIO

### ISSUE THREAD

**Issue by mischmaesch** [intridea/multi_json#156]
If we try to use JrJackson in conjunction with MultiJson in a rails app running
in a Java Servlet container (jetty) using JRuby-Rack we get the following exception.
```
Java::JavaLang::ClassCastException (org.jruby.rack.RackInput cannot be cast to org.jruby.RubyIO):
  com.jrjackson.JrJacksonRaw._parse(JrJacksonRaw.java:111)
  com.jrjackson.JrJacksonRaw.parse_str(JrJacksonRaw.java:89)
  com.jrjackson.JrJacksonRaw$INVOKER$s$1$0$parse_str.call(JrJacksonRaw$INVOKER$s$1$0$parse_str.gen)
  org.jruby.runtime.callsite.CachingCallSite.call(CachingCallSite.java:168)
  [...]
```

So `_parse` gets an `RackInput` object in `arg` which it tries to cast to `RubyIO` and fails.

Gem versions:
- jrjackson (0.2.8)
- jruby-rack (1.1.16)
- multi_json (1.10.1)

OS: Windows Server 2008 R2

We got around the problem by monkeypatching MultiJson's JrJackson adapter:
```ruby
class JrJackson
  def load(string, options={})
    if string.nil? || string.is_a?(String)
      ::JrJackson::Json.load(string, options)
    else
      ::JrJackson::Json.load(string.read, options)
    end
  end
end
```

**Comment by mbleigh** [comment]
Do other implementations handle this by default? Seems like the place to do
the read is  before it gets passed into multi JSON

**Comment by rwz** [comment]
@mbleigh we have other implementations explicitly doing `string.read` on IO objects. The problem is that JrJackson doesn't need that and in most cases is able to recognize IO automatically.

Here's the test: https://github.com/intridea/multi_json/blob/895c0771b553effbc3053a70d7f29305486c949c/spec/shared/adapter.rb#L203-L206

**Comment by rwz** [comment]
@mischmaesch I've pushed a possible fix here: https://github.com/intridea/multi_json/commit/e66f969df7b3373f7b33820c1791383678d7ec29

Could you bundle from github and check if it solves your problem?

**Comment by mischmaesch** [comment]
@rwz Yes, that does seem to help.

Thanks for the quick response!

**Comment by rwz** [comment]
Fixed by 1.11.0 release
